fem/transfer.cpp

```diff
@@ -659,44 +639,35 @@ L2ProjectionH1GridTransfer::L2ProjectionH1::L2ProjectionH1(
          // within the coarse high-order element in reference space
          emb_tr.SetPointMat(pmats(cf_tr.embeddings[ilor].matrix));
 
-         int order = fe_lor.GetOrder() + fe_ho.GetOrder() + el_tr->OrderW();
-         const IntegrationRule* ir = &IntRules.Get(geom, order);
-         M_LH_el = 0.0;
-         for (int i = 0; i < ir->GetNPoints(); i++)
-         {
-            const IntegrationPoint& ip_lor = ir->IntPoint(i);
-            IntegrationPoint ip_ho;
-            ip_tr.Transform(ip_lor, ip_ho);
-            fe_lor.CalcShape(ip_lor, shape_lor);
-            fe_ho.CalcShape(ip_ho, shape_ho);
-            el_tr->SetIntPoint(&ip_lor);
-            // For now we use the geometry information from the LOR space
-            // which means we won't be mass conservative if the mesh is curved
-            double w = el_tr->Weight() * ip_lor.weight;
-            shape_lor *= w;
-            AddMultVWt(shape_lor, shape_ho, M_LH_el);
-         }
+         ElemMixedMass(geom, fe_ho, fe_lor, el_tr, ip_tr, M_LH_el);
+
+         Array<int> dofs_lor(nedof_lor);
          fes_lor.GetElementDofs(ilor, dofs_lor);
          Vector R_row;
          for (int i = 0; i < nedof_lor; ++i)
          {
             M_LH_el.GetRow(i, R_row);
             R_el.SetRow(i, R_row.Set(ML_inv[dofs_lor[i]], R_row));
          }
+         Array<int> dofs_ho(nedof_ho);
          fes_ho.GetElementDofs(iho, dofs_ho);
          M_LH.AddSubMatrix(dofs_lor, dofs_ho, M_LH_el);
-         R->AddSubMatrix(dofs_lor, dofs_ho, R_el);
+         R.AddSubMatrix(dofs_lor, dofs_ho, R_el);
       }
    }
-}
 
-L2ProjectionH1GridTransfer::L2ProjectionH1::~L2ProjectionH1()
-{
-   delete R;
-   if (P) { delete P; }
+   // Create PCG solver
+   RTxM_LH = *TransposeMult(R, M_LH);
```

Comment:
This will leak memory

Reply:
Thanks for catching this!  Fixed in the latest commit.

Reply:
I don't think your fix will work. I would suggest making `RTxM_LH` a pointer. You could also do something like
```c++
SparseMatrix *tmp = TransposeMult(R, M_LH);
RTxM_LH = *tmp;
delete tmp;
```
but that causes an unnecessary copy.

Reply:
Sounds good.  I made `RTxM_LH` a pointer.